launcher/minecraft/mod/ShaderPack.h

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2022 Rachel Powers <508861+Ryex@users.noreply.github.com>
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2022 Rachel Powers <508861+Ryex@users.noreply.github.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "Resource.h"
+
+/* Info:
+ * Currently For Optifine / Iris shader packs, 
+ * could be expanded to support others should they exsist?
+ *
+ * This class and enum are mostly here as placeholders for validating 
+ * that a shaderpack exsists and is in the right format,
+ * namely that they contain a folder named 'shaders'.
+ *
+ * In the technical sense it would be possible to parse files like `shaders/shaders.properties` 
+ * to get information like the availble profiles but this is not all that usefull without more knoledge of the 
+ * shader mod used to be able to change settings
+ *
+ */
```

Comment:
```suggestion
/* Info:
 * Currently For Optifine / Iris shader packs, 
 * could be expanded to support others should they exist?
 *
 * This class and enum are mostly here as placeholders for validating 
 * that a shaderpack exists and is in the right format,
 * namely that they contain a folder named 'shaders'.
 *
 * In the technical sense it would be possible to parse files like `shaders/shaders.properties` 
 * to get information like the available profiles but this is not all that useful without more knowledge of the 
 * shader mod used to be able to change settings.
 */
```